### PR TITLE
Add frozen states

### PR DESCRIPTION
**Pull request by mesmith75**
Finished the other part of #1828 .

Adds `completed_frozen` and `failed_frozen` jobs status, with `j.freeze()` and `j.unfreeze()` methods. You can freeze a master job or individual subjobs.

If a subjob is frozen it cannot be resubmitted (the status transition is not allowed). If a master job is frozen none of the subjobs can be resubmitted, even if they are not frozen (the master status transition is not allowed).

### CODE REVIEW

**Review by egede — COMMENTED** [review]
Is there a status list in the GUI somewhere that should be modified as well?

**Comment by mesmith75** [comment]
Ah yes, good point. I had forgotten about the GUI. I think that will need some modification

File: ganga/GangaGUI/api/routes.py [hunk 1]
```diff
@@ -530,6 +530,10 @@ def jobs_statistics():
         # Store statistics
         for stat in ["new", "running", "completed", "failed", "killed"]:
             statistics[stat] = len(jobs.select(status=stat))
+            if stat == "completed":
+                statistics[stat] = len(jobs.select(status="completed_frozen"))
+            elif stat == "failed":
+                statistics[stat] = len(jobs.select(status="failed_frozen"))
```
Comment (egede) [inline ganga/GangaGUI/api/routes.py]: Should these not be added to the numbers for completed and failed (rather than replacing them).

Comment (mesmith75) [reply ganga/GangaGUI/api/routes.py]: yes

**Review by egede — APPROVED** [review]

**Pull request merged**